packages/cli/src/commands/init/init.ts

```diff
@@ -453,37 +418,27 @@ async function createTemplateUri(
     return options.template;
   }
 
-  // Figure out whether we should use the @react-native-community/template over react-native/template,
-  // this requires 2 tests.
-
-  // Test #1: Does the @react-native-community/template@ exist?
-  const canUseCommunityTemplate = await npmCheckPackageVersionExists(
-    TEMPLATE_PACKAGE_COMMUNITY,
-    version,
-  );
-
-  // Test #2: Does the react-native@version package *not* have a template embedded. We know that
-  //          this applies to all version before 0.75. The 1st release candidate is the minimal
-  //          version that has no template.
+  // Does the react-native@version package *not* have a template embedded. We know that this applies to
+  // all version before 0.75. The 1st release candidate is the minimal version that has no template.
   const useLegacyTemplate = semver.lt(
     version,
     TEMPLATE_COMMUNITY_REACT_NATIVE_VERSION,
   );
 
-  if (!useLegacyTemplate && !canUseCommunityTemplate) {
-    // This isn't going to succeed, stop the user and help them unblock themselves.
-    logger.warn(
-      `You've asked to install react-native@${version}, but there isn't a matching @react-native-community/template@${version}.
-
-Something has gone wrong here 😔.  If you know what you're doing, specify the exact ${chalk.bold(
-        '--template',
-      )}`,
-    );
-    process.exit(1);
-  }
-
-  if (!useLegacyTemplate && canUseCommunityTemplate) {
-    return `${TEMPLATE_PACKAGE_COMMUNITY}@${version}`;
+  if (!useLegacyTemplate) {
+    // This will use ~ to grab the latest patch version.
+    //
+    // Because we allow for many PATCHes to occur for a version of react-native, it's not possible to know
+    // (without inspecting the template's npm metadata) which template works with which react-native version.
+    // We are guaranteed that the latest version of react-native  will work with the latest version of the
+    // template on that version.
+    //
+    // Example:
+    //
+    // Suppose react-native 0.75.0 was released, and 0.74.8 was the previous version. If I ran
+    // `init --version 0.74.8` it would be guaranteed to work with the latest version of the template
+    // matching the MAJOR.MINOR, e.g. 0.74.21.
+    return `${TEMPLATE_PACKAGE_COMMUNITY}@~${version}`;
```

Comment:
Here you should recompose the version no?

As what's going to happen if version is `0.75.0-rc.0`?
What will `~0.75.0-rc.0` resolve? It should be `~0.75.0` instead.

Reply:
I'm adding a branch for `nightly` which is the only tag that doesn't play nicely with `~`.

Reply:
I've made the changes requested and tested against:

| Test | Outcome | Notes |
| -- | -- | -- |
| `node ./packages/cli/build/bin.js init TestInit --skip-install --version nightly` | ✅ |
| `node ./packages/cli/build/bin.js init TestInit --skip-install --version next` | ❌ but expected | Uses the `0.75.0-rc.0` version of the template which has the node `modules` issue |
| `node ./packages/cli/build/bin.js init TestInit --skip-install --version 0.75.0-rc.0` | ❌ but expected | Uses the `0.75.0-rc.0` version of the template which has the node `modules` issue |
| `node ./packages/cli/build/bin.js init TestInit --skip-install --version next --template @react-native-community/template@0.0.4` | ✅ | It won't build, but the project generates correctly - React Native is 1000.0.0 |
| `node ./packages/cli/build/bin.js init TestInit --skip-install  --template @react-native-community/template@0.0.4` | ✅ | It won't build, but the project generates correctly - React Native is 1000.0.0 |

The depends on react-native-community/template#30 having published artefacts.